### PR TITLE
Move wheel and fuzz into main workspace and fix CI errors

### PR DESCRIPTION
**Pull request by Rigidity**
This PR makes the following changes:
- Adds `wheel` and `fuzz` to the workspace.
- Updates the `.gitignore` file to include `target` so the generated folder isn't considered by source control in editors.
- Fixes CI permission error relating to Twine.
- Fixes CI linkage error by excluding fuzz from the workspace build (still enabled in tests).

### CODE REVIEW

**Comment by arvidn** [comment]
what do we gain by doing this?
I think there's a downside in the workflow files needing to explicitly run `clippy` and `fmt` in more workspaces (I would think we would want to move in the direction of putting everything in a single workspace).

**Comment by Rigidity** [comment]
As you said, `wheel` can't be part of the workspace and both `wheel` and `fuzz` need to have a Cargo.lock file pinning their dependencies. This achieves all of that without needing a Cargo.lock for the library.

**Comment by Rigidity** [comment]
Looks like Cargo will ignore the lock file for libraries anyway. So if you feel strongly that they should be combined into a single workspace I can do so

**Comment by Rigidity** [comment]
The main reason for this PR is the gitignore. It's unusably annoying to have my editor create target folders (with hundreds of generated files) every time I use Rust Analyzer or compile the code.

**Comment by arvidn** [comment]
> As you said, wheel can't be part of the workspace and both wheel and fuzz need to have a Cargo.lock file pinning their dependencies. This achieves all of that without needing a Cargo.lock for the library.

If we can find a way for the fuzzer and pyo3 to be able to coexist in a workspace, and use the *same* lockfile, I think that would be ideal. The versions we build the releases from really *should* be the same version we fuzz. Right now we can't guarantee that because we build the production releases from the `chia_rs` repo.

I don't think addressing your IDE issues should take precedence over that. Worst case you can reconfigure your IDE.

> It's unusably annoying to have my editor create target folders (with hundreds of generated files) every time I use Rust Analyzer or compile the code

Does it do this because the lockfile is not in `.gitignore`?

**Comment by Rigidity** [comment]
The IDE issues were from `target` not being gitignored. This absolutely should be in the gitignore file and isn't really controversial.

File: wheel/Cargo.lock
Comment (arvidn) [inline wheel/Cargo.lock:1]: I would have expected this to no longer be used, and removed from the repository. Is there a reason to leave it in?

File: wheel/Cargo.toml [hunk 1]
```diff
@@ -9,6 +9,9 @@ homepage = "https://github.com/Chia-Network/clvm_rs/"
 repository = "https://github.com/Chia-Network/clvm_rs/"
 readme = "README.md"
 
+[workspace]
+members = ["."]
```
Comment (arvidn) [inline wheel/Cargo.toml]: my understanding is that this will prevent `wheel` to be included in the main workspace.
The name of this commit moves the wheel project into its *own* workspace, which contradicts the PR title.

File: fuzz/.gitignore [hunk 2]
```diff
@@ -1,3 +1,3 @@
 target
 corpus
-artifacts
+artifacts
```
Comment (arvidn) [inline fuzz/.gitignore:2]: did you remove the newline at the end of this file? it seems a bit odd

File: fuzz/Cargo.toml [hunk 3]
```diff
@@ -5,6 +5,9 @@ authors = ["Arvid Norberg <arvid@chia.net>"]
 publish = false
 edition = "2021"
 
+[workspace]
+members = ["."]
+
```
Comment (arvidn) [inline fuzz/Cargo.toml]: this change seems redundant, and it's also not making the change suggested by the PR title

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,2 @@
-.idea/
-wasm/target
+/target
+/Cargo.lock
```
Comment (arvidn) [inline .gitignore]: I think we should follow the convention of having trailing newlines in text files

File: Cargo.lock
Comment (arvidn) [inline Cargo.lock:1]: this is also moving away from the title of this PR. this is not moving towards all tests and fuzzing using the same dependencies.
I don't believe you justify this change in the PR description.

File: wheel/.gitignore [hunk 5]
```diff
@@ -1 +0,0 @@
-target
```
Comment (arvidn) [inline wheel/.gitignore]: you just added this file in a previous commit. I think it would be better to not add it at all if you're just removing it again

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,2 @@
-/target
+target
```
Comment (arvidn) [inline .gitignore]: you've changed this file multiple times now. I think you should just keep the most recent version of the file

File: fuzz/Cargo.toml [hunk 7]
```diff
@@ -5,9 +5,6 @@ authors = ["Arvid Norberg <arvid@chia.net>"]
 publish = false
 edition = "2021"
 
-[workspace]
-members = ["."]
-
```
Comment (arvidn) [inline fuzz/Cargo.toml]: this is undoing an earlier commit. I think it would be best to just squash them

File: .github/workflows/benchmark.yml
Comment (arvidn) [inline .github/workflows/benchmark.yml:1]: I don't think this is a clear cut improvement. You add noise to the git history (`git blame` specifically) with virtually no benefit. If the formatting is important, we should enforce it with `pre-commit` (or something like that), and on CI.

File: .prettierrc [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "useTabs": false,
+  "tabWidth": 2,
+  "singleQuote": true
+}
```
Comment (arvidn) [inline .prettierrc:5]: This commits to a specific tool to maintain this formatting as well. Is this the best tool?

File: .github/workflows/build-test.yml [hunk 9]
```diff
@@ -237,7 +237,7 @@ jobs:
           components: rustfmt, clippy
 
       - name: build
-        run: cargo build --workspace
+        run: cargo build --workspace --exclude clvm_rs-fuzz
```
Comment (arvidn) [inline .github/workflows/build-test.yml:240]: I think this warrants a comment

File: .github/workflows/build-arm64-wheels.yml [hunk 10]
```diff
@@ -59,7 +59,7 @@ jobs:
 
       - name: Install Twine
         run: |
-          if [ ! -f "venv" ]; then rm -rf venv; fi
+          if [ ! -f "venv" ]; then sudo rm -rf venv; fi
```
Comment (arvidn) [inline .github/workflows/build-arm64-wheels.yml:62]: why is this necessary now? Is this an unrelated regression in the build environment?

**Review by arvidn — COMMENTED** [review]
I don't think the formatting update is an obvious improvement, but I'm not opposed (I just think it's poorly justified).

I think everything else looks good but should be squashed

**Comment by Rigidity** [comment]
Replaced by cleaner #328

**Comment by richardkiss** [comment]
> The IDE issues were from `target` not being gitignored. This absolutely should be in the gitignore file and isn't really controversial.

@Rigidity I've kind of hated `target` directories being dropped everywhere and being backed up by Time Machine on my Mac, so I added the following to my `~/.cargo/config`:

```
[build]
target-dir = "/Users/kiss/tmp/cargo-target/"
```

Now all builds go into `tmp`, which I've excluded from backup.

Also, if you add

```
[net]
git-fetch-with-cli = true
```

your updates of the cargo database will be a lot faster.
